### ISSUE TITLE
add ability to get versioning information

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#297]
**Reported by rcurtin on 2 Aug 43865191 17:46 UTC**
```
On Mon, Nov 11, 2013 at 12:31:10PM +0100, Giampaolo Bellini wrote:
> Dear all
> 
>   I'm using latest mlpack on linux for neighbor search
> 
>   I wonder if there is a way to retrieve the library version in order to
> show it in the about box of my application...
> 
>      thanks a lot
> 
>              Giampaolo
```

There are two things we should do here.

 * add a --version option to each mlpack executable (in cli.hpp or similar)
 * add a C++ function that will return a string containing the version of mlpack

There's a bit more, though; a subversion post-commit hook will need to be added to update the definition of the version string, so that it will look something like 'r15886' instead of just 'trunk' if trunk is being used.

**Comment by zoq** [comment]
Since we switched from svn to git (github) the `GetVersion()` function doesn't work anymore.

The following files are used to get the version that need to be updated:

- mlpack/CMakeLists.txt
- mlpack/src/mlpack/core/util/version.cpp
- mlpack/CMake/CreateSVNVersionHeader.cmake

For now, Ryan implemented a temporary workaround to fix the problem in 61a1064198551cb6c4e163f828cdd630715172ed which creates the version string `1.x.x`.

**Comment by rcurtin** [comment]
Fixed in 15c5546.
